packages/neuron-wallet/src/controllers/asset-account.ts

```diff
@@ -140,4 +154,68 @@ export default class AssetAccountController {
       result,
     }
   }
+
+  public async migrateAcp(params: MigrateACPParams): Promise<Controller.Response<string>> {
+    const tx = await TransactionGenerator.generateMigrateLegacyACPTx(params.id)
+
+    const txHash = await new TransactionSender().sendTx(params.id, tx!, params.password)
+
+    return {
+      status: ResponseCode.Success,
+      result: txHash,
+    }
+  }
+
+  public async showACPMigrationDialog() {
+    const walletsService = WalletsService.getInstance()
+    const currentWallet = walletsService.getCurrent()
+    const walletId = currentWallet!.id;
+
+    if (this.displayedACPMigrationDialogByWalletIds.has(walletId)) {
+      return
+    }
+
+    const syncStatus = await new SyncApiController().getSyncStatus()
+    if (syncStatus !== SyncStatus.SyncCompleted || BrowserWindow.getAllWindows().length !== 1) {
```

Comment:
To make sure the `CommandSubject` pushes the `migrate-acp` event to the main window, since the events from the `CommandSubject` are only effective in the main window but not in other windows such as the setting window.

Reply:
It seems that the command won't be emitted if the setting window is open since there're 2 windows. Actually, the command can be broadcast and only the main window will respond, other windows will ignore it.